app/team.go

```diff
@@ -142,32 +142,74 @@ func (a *App) sendTeamEvent(team *model.Team, event string) {
 	a.Publish(message)
 }
 
+func (a *App) GetSchemeRolesForTeam(teamId string) (string, string, *model.AppError) {
+	var team *model.Team
+	var err *model.AppError
+
+	if team, err = a.GetTeam(teamId); err != nil {
+		return "", "", err
+	}
+
+	if team.SchemeId != nil && len(*team.SchemeId) != 0 {
+		if scheme, err := a.GetScheme(*team.SchemeId); err != nil {
+			return "", "", err
+		} else {
+			return scheme.DefaultChannelUserRole, scheme.DefaultChannelAdminRole, nil
+		}
+	}
+
+	return model.TEAM_USER_ROLE_ID, model.TEAM_ADMIN_ROLE_ID, nil
+}
+
 func (a *App) UpdateTeamMemberRoles(teamId string, userId string, newRoles string) (*model.TeamMember, *model.AppError) {
```

Comment:
Can you think of any way to share some of the duplicated logic between `UpdateTeamMemberRoles` and `UpdateChannelMemberRoles`? It's elegantly done but in two places.

Reply:
I don't think it's worth it - they are just different enough that making it generic would probably be more complicated than it is right now. However, thanks for flagging this up, because it made me spot a nasty little bug in the implementation, which I'll shortly push a fix for to the PR!